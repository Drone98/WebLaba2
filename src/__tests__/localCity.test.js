import LocalCity from '../classes/localCity'
import React from 'react';
import renderer from 'react-test-renderer'
import '@babel/polyfill'
import SelectedCity from "./selectedCity.test";

it('Rendered local city', async () => {
    let component = await renderer.create(
        <LocalCity lat={66.66} lon={66.66}/>
    );
    let cur = component.toJSON();
    expect(cur).toMatchSnapshot();
});

it('Loading', () => {
    let component = renderer.create(
        <LocalCity lat={66.66} lon={66.66} loadState={true}/>
    );
    let cur = component.toJSON();
    expect(cur).toMatchSnapshot();
});

it('Server error', async () => {
    let axios = {
        get: () =>{
            return Promise.reject({
                response: {
                    status: 400
                }
            })
        }
    };
    let component = renderer.create(
        <div></div>
    );
    await renderer.act(() => {
        component = renderer.create(
            <LocalCity axios={axios} lat={66.66} lon={66.66}/>
        );
    });

    let cur = component.toJSON();
    expect(cur).toMatchSnapshot();
});

it('City not found', async () => {
    let axios = {
        get: () =>{
            return Promise.reject({
                response: {
                    status: 404
                }
            })
        }
    };
    let component = renderer.create(
        <div></div>
    );

    await renderer.act(() => {
        component = renderer.create(
            <LocalCity axios={axios} lat={66.66} lon={66.66}/>
        );
    });

    let cur = component.toJSON();
    expect(cur).toMatchSnapshot();
});

it('Network error', async () => {
    let axios = {
        get: () =>{
            return Promise.reject({
                response: null
            })
        }
    };
    let component = renderer.create(
        <div></div>
    );

    await renderer.act(() => {
        component = renderer.create(
            <LocalCity axios={axios} lat={66.66} lon={66.66}/>
        );
    });

    let cur = component.toJSON();
    expect(cur).toMatchSnapshot();
});
