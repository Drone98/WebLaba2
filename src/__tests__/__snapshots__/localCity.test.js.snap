// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City not found 1`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;

exports[`Loading 1`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;

exports[`Loading 2`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;

exports[`Network error 1`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;

exports[`Rendered local city 1`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;

exports[`Rendered selected city 1`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;

exports[`Server error 1`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;

exports[`Server error 2`] = `
<div
  className="loader"
>
  <b>
    Загрузка
  </b>
  <img
    alt=""
    className="loading-icon"
    src=""
  />
</div>
`;
